operator/src/main/resources/scripts/startServer.sh

```diff
@@ -188,12 +206,109 @@ checkEnv \
   AS_SERVICE_NAME || exitOrLoop
 
 trace "LOG_HOME=${LOG_HOME}"
+trace "DATA_HOME=${DATA_HOME}"
 trace "SERVER_OUT_IN_POD_LOG=${SERVER_OUT_IN_POD_LOG}"
 trace "USER_MEM_ARGS=${USER_MEM_ARGS}"
 trace "JAVA_OPTIONS=${JAVA_OPTIONS}"
+trace "KEEP_DEFAULT_DATA_HOME=${KEEP_DEFAULT_DATA_HOME}"
+trace "LINK_SERVER_DATA_DIR=${LINK_SERVER_DATA_DIR}"
+
+#
+# If DATA_HOME env variable exists than this implies override directory (dataHome attribute of CRD) specified
+# so we need to try and link the server's 'data' directory to the centralized DATA_HOME directory
+# If 'KEEP_DEFAULT_DATA_HOME' env variable is defined then we will NOT link the server's 'data' directory to the
+# centralized DATA_HOME directory and instead keep the server's 'data' directory in its default location of
+# ${DOMAIN_HOME}/servers/${SERVER_NAME}/data
+if [ ! -z ${DATA_HOME} ]; then
+  # Create $DATA_HOME directory for server if doesn't exist
+  if [ ! -d ${DATA_HOME}/${SERVER_NAME}/data ]; then
+    trace "Creating directory '${DATA_HOME}/${SERVER_NAME}/data'"
+    createFolder ${DATA_HOME}/${SERVER_NAME}/data
+  else
+    trace "Directory '${DATA_HOME}/${SERVER_NAME}/data' exists"
+  fi
+
+  # If 'LINK_SERVER_DATA_DIR' env is defined and 'KEEP_DEFAULT_DATA_HOME' environment variable is not defined then
+  # try to link server's default 'data' directory (${DOMAIN_HOME}/servers/${SERVER_NAME}/data) to $DATA_HOME/${SERVER_NAME}/data.
+  if [ ! -z ${LINK_SERVER_DATA_DIR} ] && [ -z ${KEEP_DEFAULT_DATA_HOME} ]; then
+    # if server's default 'data' directory (${DOMAIN_HOME}/servers/${SERVER_NAME}/data) does not exist than create
```

Comment:
This is 80+ of complex code that we only plan to enable experimentally, and it significantly lengthens this script, so it's a bit distracting to include it in-line.  If you agree, maybe we can hide it a bit?  Maybe move the guts of the 'if' to a function that's included at the end utils.sh, and only leave the 'if' in this script?